src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -799,6 +799,44 @@ BOOL CEEInfo::isValidStringRef (
     return result;
 }
 
+int CEEInfo::getStringLength (
+        CORINFO_MODULE_HANDLE       moduleHnd,
+        mdToken                     metaTOK)
+{
+    CONTRACTL{
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    int length = 0;
+    Module* module = GetModule(moduleHnd);
+
+    JIT_TO_EE_TRANSITION();
+
+    if (IsDynamicScope(moduleHnd))
+    {
+        length = GetDynamicResolver(moduleHnd)->GetStringLength(metaTOK);
+    }
+    else
+    {
+        DWORD dwCharCount;
+        LPCWSTR pString;
+        if (!FAILED((module)->GetMDImport()->GetUserString(metaTOK, &dwCharCount, NULL, &pString)) && (pString != nullptr))
```

Comment:
Is this overly conservative for empty strings?

Try `"".Length` ?

Reply:
Indeed! for `string.Empty` it returns NULL there so I guess `!FAILED(...)` is enough to check for errors. Going to re-run jit-diff then.